File: algebra/arithmetic_progression/clojure/arithmetic_progression.clj
;; Arithmetic Progression with n-terms
;; a, (a+d), (a+2d), (a+3d),...... (a+(n-1)d)
;;Initial Term -> a, Common Difference -> d, No: of Terms -> n

(defn nth-term [a d n]
  "gives the nth-term in a A.P"
  (+ a (* (- n 1)
          d)))

(defn sum-of-first-n [a d n]
  "gives the sum of first n-terms in a A.P"
  (* (/ n 2)
     (+ (* 2 a)
        (* (- n 1)
           d))))

;;Tests:

(assert (= 5 (nth-term 1 1 5)))          ;;1,2,3,4,5
(assert (= 27 (nth-term 0 3 10)))        ;; 0,3,6,9,12...27

(assert (= 45 (sum-of-first-n 0 1 10)))  ;; 0+1+2+..+9 = 45
(assert (= 0 (sum-of-first-n -3 1 7)))   ;; (-3)+(-2)+(-1)+0+1+2+3 = 0

File: algebra/arithmetic_progression/java/arithmetic_progression.java
// For an arithmetic progression a + (a + d) + (a + 2d) + (a + 3d) + ...
// with a being the initial term and d being the common difference
import java.util.*;
public class arithmetic_progression {

    // This function calculates the nth term
    public static int nth_term(int a, int d, int n) {
        return a + (n - 1) * d;
    }

    // This function calculates the sum of the first n terms
    public static int sum_of_first_n(int a, int d, int n) {
        return ((n) * ((2 * a) + (n - 1) * d)) / 2;
    }
    public static void main(String[] args){
        Scanner scn=new Scanner(System.in);
        System.out.println("Enter number of terms in arithmentic progression:");
        int n=scn.nextInt();
        System.out.println("Enter first term of Arithmentic progression:");
        int a=scn.nextInt();
        System.out.println("Enter the mean difference between each term:");
        int d=scn.nextInt();
        System.out.println(n+ "-th " + "term of an arithmetic progression is: " + nth_term(a,d,n)  );
        System.out.println("Sum of first "+ n + " terms is: "+ sum_of_first_n(a,d,n));

    }
}
